src/js/controller/instream-html5.js

```diff
@@ -59,15 +63,18 @@ define([
             _adModel.off();
 
             // We don't want the instream provider to be attached to the video tag anymore
-            _this.off();
+            this.off();
             if (_currentProvider) {
                 _currentProvider.detachMedia();
-                _currentProvider.resetEventListeners();
+                _currentProvider.off();
                 _currentProvider.destroy();
             }
 
             // Return the view to its normal state
             _adModel = null;
+
+            _controller.off(null, null, this);
```

Comment:
From the backbone.events documentation I see
```js
// Removes all callbacks for `context` for all events.
object.off(null, null, context);
```

This could benefit from a comment for other poor devs like myself :)